crates/rollup-boost/src/flashblocks/service.rs

```diff
@@ -203,13 +203,19 @@ impl FlashblocksService {
     pub async fn get_best_payload(
         &self,
         version: PayloadVersion,
+        payload_id: PayloadId,
     ) -> Result<Option<OpExecutionPayloadEnvelope>, FlashblocksError> {
+        // Check that we have flashblocks for correct payload
+        if *self.current_payload_id.read().await != payload_id {
```

Comment:
I like it as a safe check, just noting that this would mean that the flashblocks module missed a FCU call to reset its payload id.

Reply:
I also added best_payload cleaning on FCU with attr, otherwise it won't be cleaned